oemof_visio/energy_system_graph.py

```diff
@@ -0,0 +1,279 @@
+import os
+import graphviz
+import oemof.solph
+import logging
+
+
+def fixed_width_text(text, char_num=10):
+    """Add linebreaks every char_num characters in a given text.
+
+    Parameters
+    ----------
+    text: obj:'str'
+        text to apply the linebreaks
+    char_num: obj:'int'
+        max number of characters in a line before a line break
+        Default: 10
+    Returns
+    -------
+    obj:'str'
+        the text with line breaks after every char_num characters
+
+    Examples
+    --------
+    >>> fixed_width_text("12345", char_num=2)
+    '12\\n34\\n5'
+    >>> fixed_width_text("123456")
+    '123456'
+    """
+    # total number of characters in the text
+    text_length = len(text)
+    # integer number of lines of `char_num` character
+    n_lines = int(text_length / char_num)
+    # number of character in the last line
+    last_line_length = text_length % char_num
+
+    # split the text in lines of `char_num` character
+    split_text = []
+    for i in range(n_lines):
+        split_text.append(text[(i * char_num) : ((i + 1) * char_num)])
+
+    # I if the last line is not empty
+    if n_lines > 0:
+        if last_line_length > 0:
+            split_text.append(text[((i + 1) * char_num) :])
+        answer = "\n".join(split_text)
+    else:
+        answer = text
+    return answer
+
+
+class ESGraphRenderer:
+    def __init__(
+        self,
+        energy_system=None,
+        filepath="energy_system",
```

Comment:
> It is only for the example that it saves the file.

For me, it looks like the file is always saved. This should not be the case.

Reply:
The file is only saved if one calls the method `view` or `render`, I added the method `pipe` which returns the image in a binary format without saving it.

Reply:
`filepath` could be `None` by default and if it is `None` no file is written.

I would prefer an explicit way, but this is just my opinion.

```python
def view():
    show the plot

def write/store/save(path, suffix):
    write the the plot as an image file
```

Reply:
No file is written to system unless the user calls the "view" method (how the "view" method of the graphviz api currently works). The api also have a `pipe` method which returns binary data about the file, maybe we can use those to display the image without writing it to a file (although I never did that, it should probably technically be possible)?

Reply:
I do not like `view` for writing a file but if it comes from the package it might not be worth changing it.

Is there a file written if filepath is None (as @p-snft mentioned)? If not, this might be a way to control writing/not writing. If it raises an error we have to accept it.

Reply:
There is no file written as long as one does not call the api `view` function (I one does call and no file was provided, the api will save a file nevertheless with their own default filename)